### PR TITLE
Fix/sanitise urls

### PR DESCRIPTION
**Pull request by alexanderleegs**
This PR sanitises our axios calls using `url-template`. The endpoints which make use of user input are first sanitised with url-template.

We're making use of v2.0.8 of the library instead of the latest version, the latest version is distributed in ECMAScript module syntax, which we have issues supporting.

As this change affects the majority of our endpoints, we should run e2e tests before merging this PR.

### CODE REVIEW

**Review by kishore03109 — APPROVED** [review]
I only have concerns regarding the block of code that is no longer removed. I am assuming this removal is intentional and tested, please revoke this review if that is not the case. 

It also seems like this is a very manual heavy handed approach to solving this problem rather than some middleware/centralised approach. I am approving this since this needs to be released today + I don't have any functional blockers, but have we done some research into how other teams have tackled this? I am thinking some service that just handles the actual template + expands + string check that no one else calls axios directly accept for this service

File: src/services/db/GitHubService.ts [hunk 1]
```diff
@@ -97,19 +98,31 @@ export default class GitHubService {
...
+    if (!directoryName) {
+      const endpointTemplate = urlTemplate.parse(
+        `{siteName}/contents/{fileName}`
+      )
+      return endpointTemplate.expand({
+        siteName,
+        fileName,
+      })
+    }
+    const endpointTemplate = urlTemplate.parse(
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/services/db/GitHubService.ts:110]: Yup intended! Tested and this still works as a call to github, plus the encoding is now being done by `url-template` instead

File: src/services/db/GitHubService.ts [hunk 2]
```diff
@@ -119,11 +132,13 @@ export default class GitHubService {
     siteName: string
     directoryName: string
   }) {
-    const encodedDirPath = directoryName
-      .split("/")
-      .map((folder: string | number | boolean) => encodeURIComponent(folder))
-      .join("/")
-    return `${siteName}/contents/${encodedDirPath}`
+    const endpointTemplate = urlTemplate.parse(
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/services/db/GitHubService.ts:135]: See above!

**Comment by alexanderleegs** [comment]
> I only have concerns regarding the block of code that is no longer removed. I am assuming this removal is intentional and tested, please revoke this review if that is not the case.
> 
> It also seems like this is a very manual heavy handed approach to solving this problem rather than some middleware/centralised approach. I am approving this since this needs to be released today + I don't have any functional blockers, but have we done some research into how other teams have tackled this? I am thinking some service that just handles the actual template + expands + string check that no one else calls axios directly accept for this service

As discussed offline - we'll tackle this via enforcing a certain type of string + adding a util function to generate this type of string using url-template in a future PR. Issue has been file [here](https://linear.app/ogp/issue/ISOM-725/[refactor]-enforce-url-template-on-all-axios-calls)

**Pull request merged**